From 1617e62c09923a325663c1e2b8f34adcf5e8e290 Mon Sep 17 00:00:00 2001
From: Simon <xxm@rock-chips.com>
Date: Thu, 2 Apr 2015 17:51:53 +0800
Subject: [PATCH] rk3368: iommu: workaround for switch between vpu and hevc

vpu/hevc share the same ahb, when switch vpu and hevc,may cause
iommu error,to avoid this error,skip invalidate iommu tlb when iovmm_map

Signed-off-by: Simon <xxm@rock-chips.com>
---
 drivers/iommu/rockchip-iommu.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/iommu/rockchip-iommu.c b/drivers/iommu/rockchip-iommu.c
index 8ed83e55f3f3..005e91c1271b 100644
--- a/drivers/iommu/rockchip-iommu.c
+++ b/drivers/iommu/rockchip-iommu.c
@@ -759,6 +759,9 @@ int rockchip_iommu_tlb_invalidate(struct device *dev)
 	unsigned long flags;
 	struct iommu_drvdata *data = dev_get_drvdata(dev->archdata.iommu);
 
+	if (strstr(data->dbgname, "vpu") || strstr(data->dbgname, "hevc"))
+			return 0;
+
 	spin_lock_irqsave(&data->data_lock, flags);
 
 	if (rockchip_is_iommu_active(data)) {
-- 
2.35.3

